### PR TITLE
test(hx-accordion-panel): implement component tests

### PR DESCRIPTION
**Pull request by BY00565233**
## Description

Test Case `<hx-accordion-panel>`

image

### What are the relevant story cards/tickets? Any additional PRs or other references?

Jira: SURF-1995

## Before you request a review for this PR:

- [x] Did you add component tests for any new code?
- [x] Did you run the component unit tests via `yarn test` to ensure all tests passed?
- [x] Did you include a screenshot of the component tests?
- [ ] If you changed/added functionality, did you update the demo page and documentation?
- [ ] If needed, did you add or modify the demo test page to test the changed/added functionality?
- [x] Did you assign reviewers?
- [x] In Jira, have you linked to this PR on the ticket(s)?

### CODE REVIEW

File: src/elements/hx-accordion-panel/index.spec.js [hunk 1]
```diff
@@ -0,0 +1,148 @@
...
+            });
+
+            it('should render the Shadow Root mode open', async () => {
+                const component = /** @type { HXAccordionPanelElement } */ await fixture(template);
+                const mode = component.shadowRoot.mode;
+
+                expect(mode).to.equal('open');
+            });
+
+            it('should have a single <slot>', async () => {
```
Comment (100stacks) [inline src/elements/hx-accordion-panel/index.spec.js:51]: Please update test description:

```js
 it('should have two <slot>s', async () => {
```

File: src/elements/hx-accordion-panel/index.spec.js [hunk 2]
```diff
@@ -0,0 +1,148 @@
...
+                const component = /** @type { HXAccordionPanelElement } */ await fixture(template);
+                const shadow = component.shadowRoot;
+                const query = shadow.querySelectorAll('slot');
+                const len = query.length;
+
+                expect(len).to.be.equal(2);
+            });
+
+            it('should have an unnamed <slot>', async () => {
```
Comment (100stacks) [inline src/elements/hx-accordion-panel/index.spec.js:60]: Please add a test for the named slot - **`<slot name="header">`**.

File: src/elements/hx-accordion-panel/index.spec.js [hunk 3]
```diff
@@ -0,0 +1,148 @@
...
+                expect(queryId).to.equal(id);
+            });
+
+            it('markup should contain a <hx-icon> with an angle-down type <hx-icon>', async () => {
+                const elSelector = 'button#hxToggle > div > hx-icon';
+                const iconType = 'angle-down';
+                const component = /** @type { HXAccordionPanelElement } */ await fixture(template);
+                const shadow = component.shadowRoot;
+                const query = shadow.querySelector(elSelector);
+                const queryId = query.type;
```
Comment (100stacks) [inline src/elements/hx-accordion-panel/index.spec.js:90]: Please change variable to `queryType`.

File: src/elements/hx-accordion-panel/index.spec.js [hunk 4]
```diff
@@ -0,0 +1,148 @@
...
+    describe('test $onConnect method', () => {
+        it('should be able to open', async () => {
+            const component = /** @type {HXAccordionPanelElement} */ await fixture(template);
+            component.open = true;
+            const attr = component.hasAttribute('open');
+
+            expect(attr).to.be.true;
+        });
+
+        it('should have aria-expanded attribute', async () => {
```
Comment (100stacks) [inline src/elements/hx-accordion-panel/index.spec.js:117]: Please change test description:

```js
it('should default without an aria-expanded attribute', async () => {
```

File: src/elements/hx-accordion-panel/index.spec.js [hunk 5]
```diff
@@ -0,0 +1,148 @@
...
+    });
+
+    describe('test $onConnect method', () => {
+        it('should be able to open', async () => {
+            const component = /** @type {HXAccordionPanelElement} */ await fixture(template);
+            component.open = true;
+            const attr = component.hasAttribute('open');
+
+            expect(attr).to.be.true;
+        });
```
Comment (100stacks) [inline src/elements/hx-accordion-panel/index.spec.js:115]: You can remove this test.  It's a duplicate.

File: src/elements/hx-accordion-panel/index.spec.js (hunk 1 above)
Comment (BY00565233) [reply src/elements/hx-accordion-panel/index.spec.js]: Changes have been done. Please review

File: src/elements/hx-accordion-panel/index.spec.js (hunk 2 above)
Comment (BY00565233) [reply src/elements/hx-accordion-panel/index.spec.js:69]: Changes have been done. Please review

File: src/elements/hx-accordion-panel/index.spec.js (hunk 3 above)
Comment (BY00565233) [reply src/elements/hx-accordion-panel/index.spec.js]: Changes have been done. Please review

File: src/elements/hx-accordion-panel/index.spec.js (hunk 4 above)
Comment (BY00565233) [reply src/elements/hx-accordion-panel/index.spec.js]: Changes have been done. Please review

File: src/elements/hx-accordion-panel/index.spec.js (hunk 5 above)
Comment (BY00565233) [reply src/elements/hx-accordion-panel/index.spec.js]: Changes have been done. Please review

File: src/elements/hx-accordion-panel/index.spec.js [hunk 6]
```diff
@@ -0,0 +1,149 @@
...
+            });
+
+            it('should have an named <slot name="header">', async () => {
+                const component = /** @type { HXAccordionPanelElement } */ await fixture(template);
+                const shadow = component.shadowRoot;
+                const query = shadow.querySelectorAll('slot');
+                const len = query.length;
+
+                expect(len).to.be.equal(2);
+            });
```
Comment (100stacks) [inline src/elements/hx-accordion-panel/index.spec.js:67]: This test is the same as the one above.

Please add a test to check for the named `header` slot.

Comment (BY00565233) [reply src/elements/hx-accordion-panel/index.spec.js]: Done

**Pull request merged**